internal/framework/status/updater.go

```diff
@@ -177,70 +189,115 @@ func (upd *UpdaterImpl) updateGatewayAPI(ctx context.Context, statuses GatewayAP
 
 	upd.cfg.Logger.Info("Updating Gateway API statuses")
 
-	if upd.cfg.UpdateGatewayClassStatus {
-		for nsname, gcs := range statuses.GatewayClassStatuses {
-			upd.writeStatuses(ctx, nsname, &v1beta1.GatewayClass{}, func(object client.Object) {
-				gc := object.(*v1beta1.GatewayClass)
-				gc.Status = prepareGatewayClassStatus(gcs, upd.cfg.Clock.Now())
-			},
-			)
+	statusUpdatedCh := make(chan struct{})
+
+	go func() {
```

Comment:
Now that we are handling the context before each call to `writeStatuses` I don't think we need this goroutine either. The function will exit when the context is canceled or when it finishes all the status updates.